src/actions/file-search.js

```diff
@@ -179,15 +183,27 @@ export function traverseResults(rev: boolean, editor: Editor) {
 
 export function closeFileSearch(editor: Editor) {
   return ({ getState, dispatch }: ThunkArgs) => {
-    const query = getFileSearchQuery(getState());
-
     if (editor) {
-      const ctx = { ed: editor, cm: editor.codeMirror };
-      removeOverlay(ctx, query);
+      let codeMirror;
+      // if editor.appendToLocalElement() has not been called, editor.codeMirror
+      // throws an error.  Check for this error and if it occurs there is no
+      // need to call removeOverlay.
+      try {
+        codeMirror = editor.codeMirror;
+      } catch (err) {
+        codeMirror = null;
+      }
```

Comment:
An error is being thrown in the devtools version of code mirror [here](https://dxr.mozilla.org/mozilla-central/source/devtools/client/sourceeditor/editor.js#237).  If Editor.codeMirror is used before Editor._setup has been called it throws an error instead of just returning null or undefined.  I found a way to get the test to pass by adding  a try block to catch this error.  There may be a way to change the mochi tests to instantiate the editor but I couldn't figure that out yet.  In the debugger, the codeMirror is instantiated in the Editor component componentDidUpdate function.

Reply:
Thanks for finding the problem. I'd like to see why codeMirror might be called to soon (or too late) and find ways to prevent that)

Reply:
Is this something I should look for in the mochi test code or the application code?